### PR TITLE
Move instance data, part 1

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review. Diff from previous PR: https://github.com/vespa-engine/vespa/pull/10702/files/e9a635b9ebb4b0a44950286ce23e01d53086e263..63df4cc5b914306943a58107c922095cad691e9d

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java [hunk 1]
```diff
@@ -331,33 +333,42 @@ public void removeTenant(TenantName name) {
...
+        return readInstanceIds().filter(instanceFilter)
+                                .map(this::readInstance)
+                                .flatMap(Optional::stream)
+                                .collect(Collectors.toUnmodifiableList());
     }
 
-    public void removeApplication(ApplicationId application) {
-        curator.delete(applicationPath(application));
+    public void removeInstance(ApplicationId id) {
+        curator.delete(applicationPath(id));
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: This is a bug waiting to happen ... perhaps put a big fat warning comment here?

**Review by bratseth — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: Can do that.

**Comment by jonmv** [comment]
It ocurred to me the two-step serialised data move may lose deletes during upgrade, if one curator on old version receives a deletion request. I don't think this is likely to happen, and it wouldn't be much of an issue if it happened here. Something to keep in mind, though.

**Comment by jonmv** [comment]
Also, I think we should rename ApplicationId to InstanceId. It doesn't matter too much that some of the variables will be called (some combination of something and a stem of) `application`, because that is, after all, what they are in that context.

**Comment by jonmv** [comment]
Possible `ApplicationInstanceId`, but it's a bit long.

**Comment by jonmv** [comment]
It's also possible to have IntelliJ rename the `applicationId`, `application`, and `appId`, etc., by renaming `ApplicationId` via all these names ... o_O

**Comment by jonmv** [comment]
.

**Pull request merged**